src/vs/workbench/contrib/files/browser/views/explorerView.ts

```diff
@@ -534,12 +534,18 @@ export class ExplorerView extends ViewletPanel {
 			.sort((first, second) => second.resource.path.length - first.resource.path.length)[0];
 
 		while (item && item.resource.toString() !== resource.toString()) {
+			if (item.isDisposed) {
+				return this.onSelectResource(resource, reveal);
```

Comment:
This is cool, but can get into an endless loop if the model gets inconsistent.

Reply:
I fixed it.